Support non-standard bools in CUDA masked_scatter

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #79393
* #79392
* __->__ #79391
* #79390

The failure comes from within the `exclusive_sum` used to calculate
the selection index. So, I merged the code paths for `uint8_t` and
`bool` masks together, then use `cub::TransformIterator` to get a
proper `bool` value for the result.

peterbell10 committed:
Support non-standard bools in masked_scatter CUDA

The failure comes from within the `exclusive_sum` used to calculate
the selection index. So, I merged the code paths for `uint8_t` and
`bool` masks together, then use `cub::TransformIterator` to get a
proper `bool` value for the result.

[ghstack-poisoned]

peterbell10 committed:
Skip tracing tests for equal on "Support non-standard bools in masked_scatter CUDA"


The failure comes from within the `exclusive_sum` used to calculate
the selection index. So, I merged the code paths for `uint8_t` and
`bool` masks together, then use `cub::TransformIterator` to get a
proper `bool` value for the result.

[ghstack-poisoned]

peterbell10 committed:
Update on "Support non-standard bools in masked_scatter CUDA"


The failure comes from within the `exclusive_sum` used to calculate
the selection index. So, I merged the code paths for `uint8_t` and
`bool` masks together, then use `cub::TransformIterator` to get a
proper `bool` value for the result.

[ghstack-poisoned]

peterbell10 committed:
Skip meta tests for torch.equal on "Support non-standard bools in masked_scatter CUDA"


The failure comes from within the `exclusive_sum` used to calculate
the selection index. So, I merged the code paths for `uint8_t` and
`bool` masks together, then use `cub::TransformIterator` to get a
proper `bool` value for the result.

[ghstack-poisoned]

peterbell10 committed:
Update on "Support non-standard bools in masked_scatter CUDA"


The failure comes from within the `exclusive_sum` used to calculate
the selection index. So, I merged the code paths for `uint8_t` and
`bool` masks together, then use `cub::TransformIterator` to get a
proper `bool` value for the result.

[ghstack-poisoned]

peterbell10 committed:
Update on "Support non-standard bools in masked_scatter CUDA"


The failure comes from within the `exclusive_sum` used to calculate
the selection index. So, I merged the code paths for `uint8_t` and
`bool` masks together, then use `cub::TransformIterator` to get a
proper `bool` value for the result.

[ghstack-poisoned]

peterbell10 committed:
Update on "Support non-standard bools in masked_scatter CUDA"


The failure comes from within the `exclusive_sum` used to calculate
the selection index. So, I merged the code paths for `uint8_t` and
`bool` masks together, then use `cub::TransformIterator` to get a
proper `bool` value for the result.

[ghstack-poisoned]

peterbell10 committed:
Rebase on "Support non-standard bools in masked_scatter CUDA"


The failure comes from within the `exclusive_sum` used to calculate
the selection index. So, I merged the code paths for `uint8_t` and
`bool` masks together, then use `cub::TransformIterator` to get a
proper `bool` value for the result.

[ghstack-poisoned]

mruberry reviewed (approved):
LGTM -- @mikaylagawarecki did you want to look, too?

peterbell10 committed:
Update on "Support non-standard bools in masked_scatter CUDA"


The failure comes from within the `exclusive_sum` used to calculate
the selection index. So, I merged the code paths for `uint8_t` and
`bool` masks together, then use `cub::TransformIterator` to get a
proper `bool` value for the result.

[ghstack-poisoned]

peterbell10 committed:
Update on "Support non-standard bools in CUDA masked_scatter"


The failure comes from within the `exclusive_sum` used to calculate
the selection index. So, I merged the code paths for `uint8_t` and
`bool` masks together, then use `cub::TransformIterator` to get a
proper `bool` value for the result.

[ghstack-poisoned]

peterbell10 committed:
Rebase on "Support non-standard bools in CUDA masked_scatter"


The failure comes from within the `exclusive_sum` used to calculate
the selection index. So, I merged the code paths for `uint8_t` and
`bool` masks together, then use `cub::TransformIterator` to get a
proper `bool` value for the result.

[ghstack-poisoned]

peterbell10 commented:
@pytorchbot merge -g
